daemon/emer-machine-id-provider.c

```diff
@@ -152,31 +133,51 @@ emer_machine_id_provider_finalize (GObject *object)
   EmerMachineIdProviderPrivate *priv =
     emer_machine_id_provider_get_instance_private (self);
 
-  g_free (priv->path);
-  g_free (priv->override_path);
+  g_free (priv->tracking_id_path);
 
   G_OBJECT_CLASS (emer_machine_id_provider_parent_class)->finalize (object);
 }
 
+static void
+emer_machine_id_provider_constructed (GObject *object)
+{
+  EmerMachineIdProvider *self = EMER_MACHINE_ID_PROVIDER (object);
+  EmerMachineIdProviderPrivate *priv =
+    emer_machine_id_provider_get_instance_private (self);
+  g_autoptr(GError) local_error = NULL;
+  uuid_t uuid;
+
+  /* Check if tracking ID is already available. This make sure we do not re-create
+   * a new uuid everytime across reboots.
+   */
+  if (!emer_machine_id_provider_get_id (self, uuid))
+    {
+      if (!write_tracking_id_file (priv->tracking_id_path, &local_error))
```

Comment:
Do the tests exercise this code path? I think not because `setup` always creates the file. The most important case is trying to read the ID //without// explicitly writing it, because that will be the initial state of all real systems. (Previously we have relied on `/etc/machine-id` existing.)

Reply:
I added a new test in the suite which doesn't use `setup` so the tracking-id file is not explicitly written.